packages/editor/src/functions/uploadEnvMapBake.ts

```diff
@@ -71,46 +80,31 @@ const getScenePositionForBake = (entity?: Entity) => {
  *   which will dictate where the envmap is source from see issue #5751
  *
  * @param entity
- * @returns
+ * @returns The URL of the uploaded environment map, or null if the operation failed
  */
-
 export const uploadBPCEMBakeToServer = async (entity: Entity) => {
-  const isSceneEntity = entity === getState(EditorState).rootEntity
-
-  if (isSceneEntity) {
-    if (!hasComponent(entity, EnvMapBakeComponent)) {
-      setComponent(entity, EnvMapBakeComponent, { resolution: 1024 })
-    }
-  }
-
   const bakeComponent = getComponent(entity, EnvMapBakeComponent)
-  const position = getScenePositionForBake(isSceneEntity ? undefined : entity)
-
-  const renderer = getComponent(Engine.instance.viewerEntity, RendererComponent).renderer!
-
-  const scene = new Scene()
-
-  const cubemapCapturer = new CubemapCapturer(renderer, scene, bakeComponent.resolution)
-  const renderTarget = cubemapCapturer.update(position)
+  const bakePosition = getScenePositionForBake(entity)
+  const isSceneRootEntity = entity === getState(EditorState).rootEntity
 
-  if (isSceneEntity) scene.environment = renderTarget.texture
-
-  const envmapImageData = convertCubemapToEquiImageData(
-    renderer,
-    renderTarget.texture,
-    bakeComponent.resolution,
-    bakeComponent.resolution
-  ) as ImageData
+  const envmapImageData = generateEnvmapBake({
+    entity,
+    position: bakePosition,
+    resolution: bakeComponent.resolution
+  })
 
   const envmap = await convertImageDataToKTX2Blob(envmapImageData)
-
   if (!envmap) return null!
 
   const nameComponent = getComponent(entity, NameComponent)
   const editorState = getState(EditorState)
   const sceneName = editorState.sceneName!
   const projectName = editorState.projectName!
-  const filename = isSceneEntity ? `${sceneName}.envmap.ktx2` : `${sceneName}-${nameComponent.replace(' ', '-')}.ktx2`
+  const timestamp = Date.now()
+  // Add a cache-busting timestamp to the filename to ensure the envmap updates each bake
+  const filename = isSceneRootEntity
```

Comment:
we have cache busting already with the content hash returned on the url from static resources, which should be returned once it is uploaded. we should rework this such that once the upload is complete, we then set the EnvMapBakeComponent with the returned url with the hash

Reply:
Ah ok, existing code had the hash and search params cleared. Now updated to keep those in